simulator/quantum_trajectory_sim.py

```diff
@@ -86,3 +70,34 @@ def run(self, psi, steps):
         mask = tf.math.is_nan(tf.math.real(psi_new))
         psi_new = tf.where(mask, psi, psi_new)
         return psi_new
+
+    def measure(self, psi, Kraus, sample):
```

Comment:
I think it's better to move this `measure` elsewhere, because the trajectory simulator takes Kraus ops argument on initialization, and those Kraus ops are completely different from the Kraus ops in this method. Those ones are related to the Hamiltonian + dissipation type evolution, while this Kraus here represents an arbitrary operator which we are measuring (so in the simplest case of qubit sigma_z measurement these Kraus ops can be projectors onto qubit |0> and |1>). So I think it's better to move it elsewhere because the meaning of these two things is quite different. But "elsewhere" is tricky, I see that it doesn't logically fit anywhere :D Any suggestions?

Reply:
Yeah, I thought this location might make the most sense too - maybe instead then it's more clear if we rename the argument to `measurement_op` or something?

Reply:
yeah I think that's a good idea